npm/ux/colormap.tsx

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// essentially borrowed from https://www.npmjs.com/package/colormap
+// source are also available at https://github.com/bpostlethwaite/colormap
+
+export function ColorMap(colorsCount: number): string[] {
+  if (colorsCount <= predefinedColors.length) {
+    return predefinedColors.slice(0, colorsCount);
+  } else {
+    const cmap = jetColorMap;
+    const alpha = 1;
+
+    // map index points from 0..1 to 0..n-1
+    const indicies = cmap.map(function (c) {
+      return Math.round(c.index * colorsCount);
+    });
+
+    const steps = cmap.map((c, i) => {
+      const rgba = cmap[i].rgb.slice();
+
+      // if user supplies their own map use it
```

Comment:
How does anyone supply their own map?

Reply:
Do we need to support custom maps?
1. Map definitions can be provided as arrays for colors with 'distances' between them to set interpolations. It seems to be a complication for the API.
2. Supporting custom maps also requires some kind of error handling for situations such as the number of map intervals is greater than the number of colors need to produce.
3. Most of maps provided in colormap original samples provide shades of few number of gradients. It seems to be not applicable for the scatter chart purposes.

I do not think that providing custom color maps would be beneficial for customers.
Maybe we could supply custom colors for individual series at the Jupyter notebook scenario? E.g. draw majorana with '#00FF00' and gate-based with '#FF0000'.

What do you think @billti?

Reply:
Added support of custom colors for the Jupyter notebook scenario. Hopefully, this provides enough flexibility for users.